src/Psr/CacheItemPool/CacheItemPoolDecorator.php

```diff
@@ -269,14 +244,22 @@ public function saveDeferred(CacheItemInterface $item)
      */
     public function commit()
     {
-        $notSaved = [];
+        /** @var array<string,non-empty-array<string,CacheItem>> $groupedByTtl */
+        $groupedByTtl = [];
+        foreach ($this->deferred as $cacheKey => $item) {
+            $itemTtl                = var_export($item->getTtl(), true);
+            $group                  = $groupedByTtl[$itemTtl] ?? [];
+            $group[$cacheKey]       = $item;
+            $groupedByTtl[$itemTtl] = $group;
+        }
 
-        foreach ($this->deferred as &$item) {
-            if (! $this->save($item)) {
-                $notSaved[] = $item;
-            }
+        $notSavedItems = [];
+        foreach ($groupedByTtl as $keyValuePairs) {
+            $itemTtl         = current($keyValuePairs)->getTtl();
```

Comment:
I could use `groupedByTtl` key here aswell.
But as `null` is an invalid array-key, I'd prefer not to have an additional `if` here.